packages/frontend-test-utils/src/app/createExtensionTester.tsx

```diff
@@ -0,0 +1,227 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { Link } from '@backstage/core-components';
+import { RenderResult, render } from '@testing-library/react';
+import { createSpecializedApp } from '@backstage/frontend-app-api';
+import {
+  ExtensionDefinition,
+  IconComponent,
+  RouteRef,
+  coreExtensionData,
+  createExtension,
+  createExtensionInput,
+  createExtensionOverrides,
+  useRouteRef,
+} from '@backstage/frontend-plugin-api';
+import { MockConfigApi } from '@backstage/test-utils';
+import { JsonArray, JsonObject, JsonValue } from '@backstage/types';
+// eslint-disable-next-line @backstage/no-relative-monorepo-imports
+import { resolveExtensionDefinition } from '../../../frontend-plugin-api/src/wiring/resolveExtensionDefinition';
+// eslint-disable-next-line @backstage/no-relative-monorepo-imports
+import { signInPageComponentDataRef } from '../../../frontend-plugin-api/src/extensions/createSignInPageExtension';
+
+const TestCoreNavExtension = createExtension({
+  namespace: 'core',
+  name: 'nav',
+  attachTo: { id: 'core/layout', input: 'nav' },
+  inputs: {
+    items: createExtensionInput({
+      target: coreExtensionData.navTarget,
+    }),
+  },
+  output: {
+    element: coreExtensionData.reactElement,
+  },
+  factory({ inputs }) {
+    const NavItem = (props: {
+      routeRef: RouteRef<undefined>;
+      title: string;
+      icon: IconComponent;
+    }) => {
+      const { routeRef, title, icon: Icon } = props;
+      const to = useRouteRef(routeRef)();
+      return (
+        <li>
+          <Link to={to}>
+            <Icon />
+            {title}
+          </Link>
+        </li>
+      );
+    };
+    return {
+      element: (
+        <nav>
+          {inputs.items.map((item, index) => (
+            <NavItem
+              key={index}
+              icon={item.output.target.icon}
+              title={item.output.target.title}
+              routeRef={item.output.target.routeRef}
+            />
+          ))}
+        </nav>
+      ),
+    };
+  },
+});
+
+const TestCoreRouterExtension = createExtension({
+  namespace: 'core',
+  name: 'router',
+  attachTo: { id: 'core', input: 'root' },
+  inputs: {
+    signInPage: createExtensionInput(
+      {
+        component: signInPageComponentDataRef,
+      },
+      { singleton: true, optional: true },
+    ),
+    children: createExtensionInput(
+      {
+        element: coreExtensionData.reactElement,
+      },
+      { singleton: true },
+    ),
+  },
+  output: {
+    element: coreExtensionData.reactElement,
+  },
+  factory({ inputs }) {
+    return {
+      element: <MemoryRouter>{inputs.children.output.element}</MemoryRouter>,
```

Comment:
> [!NOTE]
> Rendering the router tracker to avoid polluting the analytics data with navigation events. Open to suggestions of doing it differently 🙂